table/block_based/block_based_table_builder.cc

```diff
@@ -473,15 +474,29 @@ struct BlockBasedTableBuilder::Rep {
           &this->internal_prefix_transform, use_delta_encoding_for_index_values,
           table_options));
     }
-    if (tbo.skip_filters) {
+    if (ioptions.optimize_filters_for_hits && tbo.is_bottommost) {
+      // Apply optimize_filters_for_hits setting here when applicable by
+      // skipping filter generation
+      filter_builder = nullptr;
+    } else if (tbo.skip_filters) {
```

Comment:
Can we also skip the "CreateFilterBlockBuilder" here if the filter_policy is nullptr?  Otherwise we end up creating a FilterBuildingContext for no good reason...